nixos/lib/test-driver/test_driver/vlan.py

```diff
@@ -24,7 +26,10 @@ def __repr__(self) -> str:
 
     def __init__(self, nr: int, tmp_dir: Path):
         self.nr = nr
-        self.socket_dir = tmp_dir / f"vde{self.nr}.ctl"
+        self.socket_tmp = tempfile.TemporaryDirectory(
+            prefix=f"vde{self.nr}.ctl", dir=tmp_dir
+        )
+        self.socket_dir = Path(self.socket_tmp.name)
```

Comment:
I would argue that teh variable `tmp_dir` should be globally set to distinct paths between multiple tests that are executed in parallel outside the sandbox.

This way we would still have all the temp files together per test, which is changed by this patch.

Reply:
Agreed. Question: do you care about the logic currently implemented by get_tmp_dir in driver.py? tempfile.gettempdir returning something that doesn't exist seems like a condition that most software wouldn't try to support and fix, and it would simplify things a lot if I could just rip that whole function off and replace with tempfile.TemporaryDirectory :)

Reply:
(I'm mostly wondering if you know of a common situation where that logic of `mkdir $TMPDIR` is actually useful -- it shouldn't be exercised in the Nix sandbox where /build is already there, and I can't imagine it happens on non-broken systems?)

Reply:
Sorry I have read what you wrote multiple times and I am a little bit confused. Can you please rephrase what you suggest to do?

Reply:
Let me rephrase: get_tmp_dir in driver.py right now doesn't make sense to me.

tempfile.gettempdir() is documented by Python to only return a *directory* that is *user writable*. If it can't find one, it returns cwd.

get_tmp_dir calls tempfile.gettempdir(), tries to mkdir the result, then checks that the result is a directory and that it's user writable. That's redundant with what tempfile.gettempdir() does.

Have I missed something that makes get_tmp_dir useful, or is ripping it out while refactoring fine?

Reply:
I understand where you're coming from.

The whole architecture is mostly inherited from the old implementation in Perl. From what it does, the function/dir could also be called "state dir" (which it is actually called in other parts of the code) because it's function is to collect in one folder structure what piles up during a test (socket files, host-guest shared dir, screenshots, ...).
This is controlled via the `TMPDIR` environment variable, which also bash scripts from within modules/virtualisation/qemu-vm.nix expect (i guess that predates even the integration test driver).
Also, this variable is documented in the test driver documentation. People will expect the driver to work in a way that they can set this variable and then all the test state shows up in there.
So we cannot just randomize it here.

From a UX perspective, it is very convenient (and i would argue that most people would expect it like this) that the test state shows up right in front of you.
If you're running multiple tests from the same folder outside the sandbox, i would still argue that you can give each invocation different TMPDIR env vars on their way.